include/flatbuffers/flatbuffers.h

```diff
@@ -1518,6 +1518,7 @@ class FlatBufferBuilder {
   // Aligns such that when "len" bytes are written, an object can be written
   // after it with "alignment" without padding.
   void PreAlign(size_t len, size_t alignment) {
+    FLATBUFFERS_ASSERT(VerifyAlignmentRequirements(alignment));
```

Comment:
for most uses of `PreAlign`, `len` is statically known, so this check is the unnecessary (or already checked by `AssertScalarT`).. not sure if its worth somehow splitting that up to avoid this runtime checking when not needed?

Reply:
This assertion is needed for public `ForceVectorAlignment()` and `ForceStringAlignment)` methods.
I moved this check into these methods.
The `PreAlign` is also public by definition but it is private by the sense. I hope no one calls this method directly.